src/client/Client.h

```diff
@@ -1417,6 +1418,78 @@ class Client : public Dispatcher, public md_config_obs_t {
     }
   };
 
+  struct C_nonblocking_fsync_state {
+    Client *clnt;
+
+    // nonblocking_fsync parms
+    Inode *in;
+    bool syncdataonly;
+    Context *onfinish;
+
+    // were local variables
+    ceph_tid_t flush_tid;
+    InodeRef tmp_ref;
+    utime_t start;
+    MetaRequest *req;
+
+    // we need to keep track of where we are
+    int progress;
```

Comment:
Maybe make this an enum so the logic is easier to folow.

Reply:
Sure